request-example.go

```diff
@@ -69,6 +75,48 @@ func (fs *root) Filewrite(r *Request) (io.WriterAt, error) {
 	return file.WriterAt()
 }
 
+func (fs *root) CheckFile(r *Request) (CheckFileResponse, error) {
+	response := CheckFileResponse{}
+	requestedAlgos := r.CheckFile.GetHashAlgoAsList()
+	var h hash.Hash
+	for _, algo := range requestedAlgos {
```

Comment:
I do not understand the purpose of ranging through the `requestedAlgos` when you will only ever pick one. Does the standard define that it will only ever respond with a single algorithm in response? If it does, then what does the standard say about how we should go about picking the proper response algorithm?